guarddog/analyzer/sourcecode/code-execution.yml

```diff
@@ -79,6 +79,10 @@ rules:
           - pattern: popen($ARG1, ...)
           - pattern: popen([..., $ARG1, ...], ...)
 
+          # distutils spawn
+          - pattern: spawn($ARG1, ...)
+          - pattern: $OBJ.spawn($ARG1, ...)
```

Comment:
would this trigger on any `xxx.spawn()` call? might cause some FPs (?)

Reply:
Yeah, I think that's right.  We might get some FPs occasionally, but I think relative to the simplicity of the new rule it's an acceptable trade-off.  It's also in keeping with other patterns in this rule (e.g., we have just `system(...)` in here).

Reply:
We could use [symbolic-propagation](https://semgrep.dev/docs/writing-rules/experiments/symbolic-propagation) feature and specify `distutils.spawn(...)`, this will mitigate that.
FYI, symbolic propagation would take care tracking things like 
```
from distutils import spawn as whatever 
whatever(...)
```

Reply:
We don't find any FPs against the top 500 packages as-is, but I like this feature.  I'll add it in and test.

Reply:
After some testing, it seems our use case with `distutils` and `spawn`is a bit too complex for what `symbolic-propagation` can handle.  Based on testing, we shouldn't expect too many FPs from the simpler form of the rule.